flojoy/flojoy_cloud.py

```diff
@@ -0,0 +1,409 @@
+            assert dc["type"] == "Grayscale", "dataContainer type does not match."
+            return dc
+
+        @validator("dataContainer")
+        def keys_must_match(cls, dc):
+            assert "m" in dc, 'dataContainer does not contain "m" dataset.'
+            assert isinstance(dc["m"], list), '"m" dataset is not a list'
+            return dc
+
+    class ScalarModel(DefaultModel[DataC], Generic[DataC]):
+        data: Optional[DataC]
+
+        @validator("dataContainer")
+        def type_must_match(cls, dc):
+            assert dc["type"] == "Scalar", "dataContainer type does not match."
+            return dc
+
+        @validator("dataContainer")
+        def keys_must_match(cls, dc):
+            assert "c" in dc, 'dataContainer does not contain "m" dataset.'
+            # int as well just in case?
+            assert isinstance(dc["c"], float), '"c" dataset is not a float'
+            return dc
+
+    class ImageModel(DefaultModel[DataC], Generic[DataC]):
+        data: Optional[DataC]
+
+        @validator("dataContainer")
+        def type_must_match(cls, dc):
+            assert dc["type"] == "Image", "dataContainer type does not match."
+            return dc
+
+        @validator("dataContainer")
+        def keys_must_match(cls, dc):
+            assert "r" in dc, 'dataContainer does not contain "r" dataset.'
+            assert "g" in dc, 'dataContainer does not contain "r" dataset.'
+            assert "b" in dc, 'dataContainer does not contain "r" dataset.'
+            assert isinstance(dc["r"], list), '"r" dataset is not a list'
+            assert isinstance(dc["g"], list), '"g" dataset is not a list'
+            assert isinstance(dc["b"], list), '"b" dataset is not a list'
+            return dc
+
+    dc_type = response["dataContainer"]["type"]
+    match dc_type:
+        case "OrderedPair":
+            try:
+                OrderedPairModel.parse_obj(response)
+            except ValidationError as e:
+                print(e)
+        case "OrderedTriple":
+            try:
+                OrderedTripleModel.parse_obj(response)
+            except ValidationError as e:
+                print(e)
+        case "DataFrame":
+            try:
+                DataFrameModel.parse_obj(response)
+            except ValidationError as e:
+                print(e)
+        case "Matrix":
+            try:
+                MatrixModel.parse_obj(response)
+            except ValidationError as e:
+                print(e)
+        case "Grayscale":
+            try:
+                GrayscaleModel.parse_obj(response)
+            except ValidationError as e:
+                print(e)
+        case "Scalar":
+            try:
+                ScalarModel.parse_obj(response)
+            except ValidationError as e:
+                print(e)
+        case "Image":
+            try:
+                ImageModel.parse_obj(response)
+            except ValidationError as e:
+                print(e)
```

Comment:
So you just want to apply them like 

OrderedPairModel.parse_obj(response)
OrderedTripleModel.parse_obj(response)
DataFrameModel.parse_obj(response)... etc. 

Will that work? It will throw 6 errors every time for the models that don't match the DC

Reply:
Yepp, just let it throw. By printing it this I am pretty sure we are losing the stacktrace of the error. Tho if the error it throws is non-readable, then we can throw a custom one in the except block but still forward the stack trace. For now just let it throw is good enough

Reply:
> So you just want to apply them like
> 
> OrderedPairModel.parse_obj(response)
> 
> OrderedTripleModel.parse_obj(response)
> 
> DataFrameModel.parse_obj(response)
> 
> ... etc.
> 
> Will that work? It will throw 6 errors every time for the models that don't match the DC

I mean keep the match statement, but just don't try catch

```python
match dc.type:
    case "OrderedPair":
        OrderedPairModel.parse_obj(response)
    case "OrderedTriple":
        OrderedTripleModel.parse_obj(response)
    case "DataFrame":
        DataFrameModel.parse_obj(response)
    ...
```

Reply:
Exactly ^